pytravharv/store.py

```diff
@@ -0,0 +1,58 @@
+import logging
+from datetime import datetime
+from typing import List
+
+from pyrdfj2 import J2RDFSyntaxBuilder
+from pyrdfstore import RDFStore
+from rdflib import Graph
+from rdflib.plugins.sparql.processor import Result
+
+log = logging.getLogger(__name__)
+
+
+class TargetStoreAccess:
```

Comment:
feels like some pattern, but can´t pin it down - something between `Decorator` (but with added methods / functionality) - `Adapter` (but not really a case of incompatible things) or `Facade` (but not really needing to hide a lot)

see also https://refactoring.guru/design-patterns

sticking to Decorator for now, as I think adding functionality and api calls is not prevented in the pattern

What happens here, is that you want to add some functionality to the RDFStore store to help out in the work TravHarv needs to do, however, you don't want to resort to subclassing aal RDFStore implementations to achieve that (and many more as future implementations could emerge) -- so you subclass the interface and wrap the implmentation:


```
def RDFStoreDecorator(RDFStore):  # subclass the original interface
    def __init__(self, store: RDFStore):
        self._core = store

   def method_x_from_rdf_store(self, **kwargs):  # just wrap *all* the methods defined in the original interface
        return self._core.method_x_from_rdf_store()


def TravHarvRDFStoreDecorator(RDFStoreDecorator):  # to be one of those decorators 
    def __init__(self, store: RDFStore, ...):
        super().__init__(store)
        # you could handle extra params

    def new_functionality(self, ..):
        # whatever you need
        # don't be tempted to call self._core.method_x_from_rdf_store() though, 
        # rather use self.method_x_from_rdf_store() to let the decorator do its magic
```

You might consider blending in the GraphTaskNameMapper on this level too)

Also, one could argue that RDFStoreDecorator should be made part of the pyrdfstore package as to support this pattern from the base.

Reply:
mentioned in #20